dask/array/reductions.py

```diff
@@ -410,8 +409,7 @@ def moment_chunk(A, order=2, sum=chunk.sum, numel=numel, dtype='f8', **kwargs):
     n = numel(A, **kwargs).astype(np.int64)
     u = total / n
     diff = A - u
-    todense = todense_lookup.dispatch(type(diff))
-    xs = [sum(todense(A - u)**i, dtype=dtype, **kwargs) for i in range(2, order + 1)]
+    xs = [((A - u)**i).sum(dtype=dtype, **kwargs) for i in range(2, order + 1)]
```

Comment:
@mrocklin this is the line that causes `nanvar` test to fail. Please note that `sum()` here is not Python's nor NumPy's `sum()`, but it's one of `moment_chunk()`'s arguments. I'm now wondering third things:

1. Should we rename the `sum` argument to something like `sum_func` to avoid confusion?
1. Is it possible there's a bug in the `sum()` function in `dask/array/reductions.py`? I don't think they should give us different results anyway.
1. Did we accidentally changed behavior in some reduction functions when `sum()` was replaced by `_concatenate2().sum()`?